### ISSUE TITLE
lat/lng updating script for agencies

### ISSUE THREAD

**Issue by josh-chamberlain** [Police-Data-Accessibility-Project/data-sources-app#246]
### Context

related to https://github.com/Police-Data-Accessibility-Project/data-source-map/issues/9#issuecomment-2059638208

### Requirements

This script should use either mapbox or google API to 

1. create a script to generate lat/lng for agencies where they are `blank`
    - GitHub Actions; not automated, unless we want to do this in small batches for any reason
    - We can try using one of those APIs to search by ZIP code or agency name, always using state and county to make the search more precise and disambiguate similar place names  
    - This will not work for some agencies, but we do not need these to be super precise. generating a lat/lng centered on the municipality/county/state is fine
       - `jurisdiction_type` will be helpful for determining `county` `state` `aggregated`

2. update our database via API endpoint
   - may need to create or modify the endpoint

When we run it the first time, we'll need to make a lot of calls for missing lat/lng—but we don't add agencies very often any more.

### Docs
- API docs

**Comment by mbodeantor** [comment]
@josh-chamberlain Not sure it matters too much which service we use, but since we have stuff on a google account already, might make billing slightly easier to go with that. I would just need an API key once you activate one of those.

**Comment by mbodeantor** [comment]
@josh-chamberlain if we're not worried about being super accurate, most of those agencies missing lat/lng (3475 of 5777) have a county name that we could use to assign lat/lng

**Comment by josh-chamberlain** [comment]
we do have mapbox too for the work Joshua G is doing!

**Comment by mbodeantor** [comment]
@josh-chamberlain using lat/lng for zip codes and counties for the agencies that have them dropped the number of agencies without lat/lng to 1873

Once we have an API key for a geocoding service I can fill the rest

**Comment by josh-chamberlain** [comment]
@mbodeantor I DM'd you one from mapbox

**Comment by mbodeantor** [comment]
@josh-chamberlain Here's a sample of the remaining agencies without lat/lng run through the search and then geocode mapbox endpoints. Already seems like some assumptions we could make around school district police departments for example but we might want to have manual review for the bulk.

[agencies_with_lat_lng.csv](https://github.com/Police-Data-Accessibility-Project/data-sources-app/files/15119762/agencies_with_lat_lng.csv)

**Comment by josh-chamberlain** [comment]
@mbodeantor nice, looks good! falling back to something general is OK too.

**Comment by josh-chamberlain** [comment]
moving this back to TODO—Marty's no longer able to work on this but I added the API key as a variable, ping me if you need it or have any questions.

**Comment by maxachis** [comment]
@josh-chamberlain This might be useful to have as another automation job, potentially. We already have the endpoint for updating agencies, so it'd be a matter of retrieving agencies with missing lat/lng and then submitting them. 

I don't know how Marty did this, but we'd probably want to make sure the logic is within our repository, so others can follow up on it.

Additionally, we'll want to make note of which agencies we've previously tried to find geocoding information for but failed. I don't know how common that will be, since we require locational information, but it's worth taking into account.

**Comment by josh-chamberlain** [comment]
@maxachis yes, this would be a good automation! A small one. We have mapbox credentials already, so it should be simple to make a new little script!

**Comment by maxachis** [comment]
Marking this as `post-v1` because it'll be easier to do once we've got the prod API up and running!

**Comment by maxachis** [comment]
@josh-chamberlain Noting this as being connected to #577 . Basically, if we add coordinates to locations, we don't need to add coordinates to agencies, and vice-versa. So one of these issues can be closed.
